components/shared/net/lib.rs

```diff
@@ -659,22 +663,43 @@ impl FetchThread {
                         .core_resource_thread
                         .send(CoreResourceMsg::Cancel(request_ids));
                 },
+                ToFetchThreadMessage::Exit => break,
             }
         }
     }
 }
 
 static FETCH_THREAD: OnceLock<Sender<ToFetchThreadMessage>> = OnceLock::new();
 
+/// Starts a fetch thread,
+/// and returns the join handle to it.
+pub fn start_fetch_thread(core_resource_thread: &CoreResourceThread) -> JoinHandle<()> {
+    let (sender, join_handle) = FetchThread::spawn(core_resource_thread);
+    FETCH_THREAD
+        .set(sender)
+        .expect("Fetch thread should be set only once on start-up");
+    join_handle
+}
+
+/// Send the exit message to the background thread,
+/// after which it can be joined on.
+pub fn exit_fetch_thread() {
```

Comment:
Remark: The doc could clarify that it doesn't join, only signals the exit and the caller must join() the handle themselves.

Reply:
Clarified.